components/admin_console/custom_enable_disable_guest_accounts_setting.tsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+import ConfirmModal from 'components/confirm_modal.jsx';
+
+import BooleanSetting from './boolean_setting';
+
+type Props = {
+    id: string;
+    value: boolean | string;
+    onChange: (id: string, value: any) => void;
+    disabled?: boolean;
+    setByEnv: boolean;
+}
+
+type State = {
+    showConfirm: boolean;
+}
+
+export default class CustomEnableDisableGuestAccountsSetting extends React.Component<Props, State> {
+    public constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            showConfirm: false,
+        };
+    }
+
+    public handleChange = (id: string, value: boolean, confirm?: boolean) => {
+        if (!value && !confirm) {
+            this.setState({showConfirm: true});
+        } else {
+            this.props.onChange(id, value);
+        }
+    };
+
+    public render() {
+        const label = (
+            <FormattedMessage
+                id='admin.guest_access.enableTitle'
+                defaultMessage='Enable Guest Access: '
+            />
+        );
+        const helpText = (
+            <FormattedMarkdownMessage
+                id='admin.guest_access.enableDescription'
+                defaultMessage='When true, external guest can be invited to channels within teams. Please see [Permissions Schemes](../user_management/permissions/system_scheme) for which roles can invite guests.'
+            />
+        );
+
+        let value;
+        if (typeof this.props.value === 'string') {
+            value = this.props.value === 'true';
+        } else {
+            value = this.props.value;
+        }
```

Comment:
In cases like this I am a fan of the ternary operator:
```js
const value = (this.props.value === 'true' || this.props.value === true) ? true : false;
```

Reply:
This is no longer needed